textpattern/publish/taghandlers.php

```diff
@@ -3600,33 +3598,20 @@ function image_info($atts)
     $validItems = array('id', 'name', 'category', 'category_title', 'alt', 'caption', 'ext', 'author', 'w', 'h', 'thumb_w', 'thumb_h', 'date');
     $type = do_list($type);
 
-    $from_form = false;
-
-    if ($id) {
-        $thisimage = imageFetchInfo('id = '.intval($id));
-    } elseif ($name) {
-        $thisimage = imageFetchInfo("name = '".doSlash($name)."'");
-    } else {
-        assert_image();
-        $from_form = true;
-    }
-
     $out = array();
-    if ($thisimage) {
-        $thisimage['category_title'] = fetch_category_title($thisimage['category'], 'image');
+    if ($imageData = imageFetchInfo($id, $name)) {
+        $imageData['category_title'] = fetch_category_title($imageData['category'], 'image');
 
         foreach ($type as $item) {
             if (in_array($item, $validItems)) {
-                if (isset($thisimage[$item])) {
+                if (isset($imageData[$item])) {
                     $out[] = ($escape == 'html') ?
-                        txpspecialchars($thisimage[$item]) : $thisimage[$item];
+                        txpspecialchars($imageData[$item]) : $imageData[$item];
                 }
+            } else {
+                trigger_error(gTxt('invalid_attribute_value', array('{name}' => $item)), E_USER_NOTICE);
             }
         }
-
-        if (!$from_form) {
-            $thisimage = '';
```

Comment:
This is great stuff, Jeff, thanks.

Do you happen to know what the point of this conditional is? I always assumed it was in case you used the tag in a loop scenario, that it resets `$thisimage` so it's not left populateed after the tag completes, therefore leading conditionals such as `<txp:if_thumbnail>` to return incorrect information if they were used outside image context at a later point on the page. Or, maybe, it's to make sure that the next image in a `<txp:images>` loop is "clean". I'm not entirely sure. Either way, it always struk me as odd that `$thisimage` is an array but it's reinitialised to an empty string instead of an empty array.

Do you think it's important enough to keep it in these tags? Or does your refactorisation negate the need for it?